File: 백준/Silver/13414. 수강신청/수강신청.py
import sys
input = sys.stdin.readline

K, L = map(int,input().split())
dic = {}
for i in range(L):
    dic[input().strip()] = i
result = sorted(dic.items(), key=lambda x : x[1])
for i in result[:K]:
    print(i[0])

File: 백준/Silver/13414. 수강신청/수강신청.swift
import Foundation

let input = readLine()!.split(separator: " ").map { Int($0)! }
let (k, l) = (input[0], input[1])
var dic = [String: Int]()
var count = 0
var printCnt = 0

for _ in 0..<l {
    let person = readLine()!
    count += 1
    dic[person] = count
}
let answer = dic.sorted { $0.value < $1.value }
for ele in answer {
    print(ele.key)
    printCnt += 1
    if printCnt == k {
        break
    }
}